optimade/validator/utils.py

```diff
@@ -67,6 +68,98 @@ def print_success(string, **kwargs):
     print(f"\033[92m\033[1m{string}\033[0m", **kwargs)
 
 
+@dataclasses.dataclass
+class ValidatorResults:
+    """ A dataclass to store and print the validation results. """
+
+    success_count: int = 0
+    failure_count: int = 0
+    internal_failure_count: int = 0
+    optional_success_count: int = 0
+    optional_failure_count: int = 0
+    failure_messages: List[Tuple[str, str]] = dataclasses.field(default_factory=list)
+    internal_failure_messages: List[Tuple[str, str]] = dataclasses.field(
+        default_factory=list
+    )
+    optional_failure_messages: List[Tuple[str, str]] = dataclasses.field(
+        default_factory=list
+    )
+    verbosity: int = 0
+
+    def add_success(self, summary: str, success_type: Optional[str] = None):
+        """Register a validation success to the results class.
+
+        Parameters:
+            summary: a summary of the success to be printed
+            success_type: either `None`, `"optional"` depending on the
+                type of the check.
+
+        """
+        success_types = (None, "optional")
+        if success_type not in success_types:
+            raise RuntimeError(
+                f"`success_type` must be one of {success_types}, not {success_type}."
+            )
+
+        if success_type is None:
+            self.success_count += 1
+        elif success_type == "optional":
+            self.optional_success_count += 1
+
+        message = f"✔: {summary}"
+        pretty_print = print if success_type == "optional" else print_success
+
+        if self.verbosity > 0:
+            pretty_print(message)
+        elif self.verbosity == 0:
+            pretty_print(".", end="", flush=True)
```

Comment:
This could also be `if self.verbosity: ... else: ...`, but being explicit here might be the better choice (i.e., this is just an empty comment...)

Reply:
I think there's still an argument to be made for switching around our verbosities, so that 0 literally prints nothing. Will be much easier to change in future if we maintain explicit integers in the code as they are (or even better, switch to an enum)